{% extends "base.html" %}
{% block title %}
BAliBASE questions
{% endblock %}
{% block additional_imports %}
<link rel="stylesheet" href="{{ url_for('static', filename='css/disprot_examples.css') }}">
{% endblock %}
{% block content %}
<div class = "panel-group">
  <div class="panel panel-default" style="width=100%;" id="panel0">
    <div class="panel-heading">
      <div class="panel-title">
        <strong>BAliBASE</strong>
      </div>
    </div>
    <div class="panel-body panel-body-custom">
      <div class="row">
        <div class="col-xs-12 col-md-6" style="overflow:auto;">
          <p align="justify">
          Correct motif alignment is an extremely difficult task to perform,
          even by manual curation. There are situations,
          where the 'correct' solution is not obvious or it does not exist
          at all. 
          Therefore, it comes to us as no surprise that there are alignments
          in the BAliBASE benchmark
          where ultimate solutions seem euqally correct.
          We provide here several examples (along with two examples
          from BAliBASE for structured proteins) to illustrate ambiguity
          that is intrinsic to protein alignment.
          <br>
          Two of these problems were explained to us by one of the
          BAliBASE creators, prof. Julie Thompson. (We include the answers in the text)
          <br>
          We found many examples where the BAliBASE creators overruled the ELM
          motif assignment. Even though KMAD often agrees with BAliBASE
          curators one should actually manually validate all cases where it
          does not. In summary, we believe that problems associated 
          with annotation and curation are responsible for a few percent of 
          the scores. Given that the methods tested differ by at most a 
          few percent too we have to conclude that all methods tested
          perform equally well considering this margin of error.
          <hr>
          Most of the following alignments are BAliBASE reference alignments 
          (unless stated otherwise), where the bright orange residues are
          residues included in the TP/FN motif instances by BAliBASE, and
          purple residues are the segments that match the ELM pattern.
          </p>
        </div>
      </div>
    </div>
  </div>
  <div class="panel panel-default" style="width=100%;" id="panel0">
    <div class="panel-heading">
      <div class="panel-title">
        <strong>How is it determined which motifs are true positives?</strong>
      </div>
    </div>
    <div class="panel-body panel-body-custom">
      <div class="row" style="height:50px;">
        <div class="col-xs-12 col-md-12" style="overflow:auto;">
          <div class="row">
            <div class="col-xs-12 col-md-6">
              <strong>Alignment BOX230 from the reference 9, subset RV921</strong>
              <br>
              <strong>Motif:</strong>
              <a href="http://elm.eu.org/elms/LIG_FHA_2">LIG_FHA_2</a>
              <br>
              <strong>Regex:</strong> "..(T)..[DE]."
              <br>
              <p align="justify">
              <strong>1.</strong>The LIG_FHA_2 motif is annotated only in 
              the first sequence, and it is predicted in the other sequences.
              When there are multiple occurences of the motif in other 
              sequences how is it decided
              which of them should be aligned with the one in the top sequence? 
              <br>
              <strong>2.</strong> Why is the motif in the 5th sequence shifted 
              to the left?
              If we assume that all of them are true positives, shouldn't they 
              all be lined up?
              <br>
              <strong>3.</strong>
              Concerning the false negative motif in the 9th sequence.
              Why do you assume that the prediction (or actually lack of 
              prediction)
              is wrong and it's not just a case of a motif changing it's 
              position?
              Motifs very often 'move around' in the disordered regions.
              <br>
              </p>
              <hr>
              <strong>Answers:</strong>
              <div style="background-color: #f5f5f5; border: 1px solid #ccc;
                border-radius: 4px; color: #333; display: block;">
                <code>
                  <div style="margin-left:5px;margin-right:5px;
                    padding-top:2px;padding-bottom:2px;">
                    <p align="justify">
                    1. We chose the motif that is aligned with the annotated
                    sequence based on the sequence conservation score of the 
                    motifs and their relative positions in the sequences with 
                    respect to other conserved blocks. For example, the 
                    annotated 
                    motif is ESTDEES. For Q6GR07, there are 2 possible motifs:  
                    GSTDEEQ and SKTGKEH, that appear close to the annotated 
                    motif in the alignment. We chose GSTDEEQ because it has 
                    a higher sequence identity with the annotated motif.
                    <br>

                    2. The motif defined in ELM is ..(T)..[DE], so in Q1DTN6, 
                    there are 2 overlapping occurrences of the motif: DETATES 
                    and TATESDR. We chose the second one to avoid introducing 
                    a new gap in the alignment that was not just by the
                    sequence identity with the first motif.
                    </p>
                  </div>
                </code>
              </div>
              <br>
              <br>
            </div>
          </div>
          <div class="row">
            <div class="col-xs-12 col-md-12">
              <p align ="justify">
              BAliBASE alignment (RV921_BOX230) with scored motifs highlighted
              (only the one in the first sequence is annotated)
              </p>
              <br>
              <iframe name="alignment_frame" src="{{ url_for('static',
                filename="alignments/balibase_issues/RV921_BOX230_bali_ref_motifs.html") }}"
                width="100%"></iframe>
              <br>
              BAliBASE alignment with all instances of the LIG_FHA_2 motif highlighted
              <br>
              <iframe name="alignment_frame" src="{{ url_for('static',
                filename="alignments/balibase_issues/RV921_BOX230_bali_predicted_motifs.html") }}"
                width="100%"></iframe>
              <br>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
<div class="panel panel-default" style="width=100%;" id="panel0">
  <div class="panel-heading">
    <div class="panel-title">
      <strong>How is the alignment curated when there are no structures available for any of the sequences? (BB11037)</strong>
    </div>
  </div>
  <div class="panel-body panel-body-custom">
    <div class="row" style="height:50px;">
      <div class="col-xs-12 col-md-12" style="overflow:auto;">
        <strong>Alignment BB11037 from reference 1 (RV911)</strong>
        <br>
        <div class="row">
          <div class="col-xs-12 col-md-6">
            <p align="justify">
            For any of the sequences from this alignment there isn't any structure available.
            The closest structure was found for the 4th sequence (YB00_YEAST),
            with a very high e-value of 3e-04. And even this structure doesn't
            cover the area where the sequences are aligned.
            <br>
            <strong>1.</strong> How was this alignment determined?
            <br>
            <strong>2.</strong> How where the 'coreblocks' determined?
            <br>
            <hr>
            </p>
          </div>
        </div>
        Reference BAliBASE alignment (highlighted coreblocks are the parts of the alignment for which BAliBASE gives points)
        <br>
        <iframe name="alignment_frame" src="{{ url_for('static',
          filename="alignments/balibase_issues/BB11037_coreblocks.html") }}"
          width="100%"></iframe>
        <br>
        The closest structure was found for the YB00_YEAST sequence (BLAST e-value: 3e-04),
        so first - it's very distant, second - it doesn't even cover the "coreblocks" area
        <br>
        <iframe name="alignment_frame" src="{{ url_for('static',
          filename="alignments/balibase_issues/BB11037_closest_structure_aln.html") }}"
          width="100%"></iframe>
        <br>
      </div>
    </div>
  </div>
</div>
<div class="panel panel-default" style="width=100%;" id="panel0">
  <div class="panel-heading">
    <div class="panel-title">
      <strong>BBS12034</strong>
    </div>
  </div>
  <div class="panel-body panel-body-custom">
    <div class="row" style="height:50px;">
      <div class="col-xs-12 col-md-12" style="overflow:auto;">
        <p align="justify">
        <strong>Alignment BB12034 from reference 1 (RV912)</strong>
        <br>
        <div class="row">
          <div class="col-xs-12 col-md-6">
            <p align="justify">
            BAliBASE is said to use SAP structure alignment as the base for
            the sequence alignment, but there are some discrepancies between the
            SAP and BAliBASE alignments. 
            T-Coffee Expresso uses SAP to produce a structure based alignment
            and the result is also different then the reference alignment.
            For example the "DRTG"
            segment is at position 33 in the Expresso and SAP alignments,
            while in the reference alignment it is shifted to the left.
            <br>
            <strong>1.</strong> Why was it decided that this segment should
            be shifted to the left, although it is aligned differently in the
            structure alignment?
            <hr>
            </p>
            <strong>Answer:</strong>
            <br>
            <div style="background-color: #f5f5f5; border: 1px solid #ccc;
              border-radius: 4px; color: #333; display: block;">
              <code>
                <div style="margin-left:5px;margin-right:5px;
                  padding-top:2px;padding-bottom:2px;">
                  <p align="justify">
                  The SAP alignments were manually refined to align residues with
                  known functions. In this case, in 1aiq, the motif DRTG contains 
                  a residue R in contact with a ligand (according to the
                  <a href="https://www.ebi.ac.uk/thornton-srv/databases/cgi-bin/pdbsum/GetPage.pl?pdbcode=1aiq&template=protein.html&r=wiring&l=1&chain=A">pdb sum website</a>). 
                  The same function is defined for 1b02 in the motif DVRTK.
                </div>
                </p>
              </code>
            </div>
            <br>
            <p align="justify">
            While we agree that the two arginines in contact with a ligand 
            should be aligned lined up together it is still not clear to us 
            why aren't they includeed in the coreblocks. We have knowledge 
            about their importance and about them being equivalent, but it 
            is not penalized if a method does not align them properly.
            </p>
            <br>
          </div>
        </div>
        <br>
        <br>
        Structure based KMAD (structure alignment performed with Dali)
        <br>
        <iframe name="alignment_frame" src="{{ url_for('static',
          filename="alignments/balibase_issues/BB12034_kmad40.html") }}"
          width="100%"></iframe>
        <br>
        <br>
        Expresso (structure alignment performed with SAP)
        <br>
        <iframe name="alignment_frame" src="{{ url_for('static',
          filename="alignments/balibase_issues/BB12034_expresso.html") }}"
          width="100%"></iframe>
        <br>
        BAliBASE refrence alignment
        <br>
        <iframe name="alignment_frame" src="{{ url_for('static',
          filename="alignments/balibase_issues/BB12034_bali.html") }}"
          width="100%"></iframe>
        <br>
        SAP structure alignment
        <br>
        <iframe name="alignment_frame" src="{{ url_for('static',
          filename="alignments/balibase_issues/BB12034_sap.html") }}"
          width="100%"></iframe>
        <strong>Ligands</strong>
        <br>
        Interactions of 1AIQ with the ligand (UMP, 2'-Deoxyuridine 5'-Monophosphate)
        <br>
        <img src="{{ url_for('static',
        filename="alignments/balibase_issues/1aiq_ligand.png") }}" >
        <br>
        Interactions of 1B02 with the ligand (UFP-C2F; 
        UFP: 5-Fluoro-2'-Deoxyuridine-5'-Monophosphate;
        C2F: 5-Methyl-5,6,7,8-Tetrahydrofolic acid)
        <br>
        <img src="{{ url_for('static',
        filename="alignments/balibase_issues/1b02_ligand.png") }}" >
        </p>
      </div>
    </div>
  </div>
</div>
<div class="panel panel-default" style="width=100%;" id="panel0">
  <div class="panel-heading">
    <div class="panel-title">
      <strong>RV913_BOX158</strong>
    </div>
  </div>
  <div class="panel-body panel-body-custom">
    <div class="row" style="height:50px;">
      <div class="col-xs-12 col-md-12" style="overflow:auto;">
        <div class="row">
          <div class="col-xs-12 col-md-6">
            <p align="justify">
            This example shows that BAliBASE motif score might not always
            be a relevant measure of motif alignment.
            The motif that BAliBASE scores for here is a KDEL motif,
            described by regular expression: 
            "[KRHQSAP][DENQT]EL$". Here we present 3 alignment excerpts,
            with motif scores 0.37, 0.88, and 0.78. Each alignment is shown
            in two versions - with BAliBASE motifs highlighted (orange) and 
            with KDEL pattern matches highlighted (purple). 
            In the alignment 1, which gets the worst score, the actual KDEL
            motifs are aligned perfectly. The best scoring one -
            alignment 2 aligns incorrectly one residue. In alignment 3,
            which also gets quite a good score, the motif residues are
            aligned with each other, but the whole motif is so spread out
            because of the gaps that it is hardly possible for it to be 
            biologically relevant.
            <br>
            There is no evidence of KDEL (or KDEL equivalent) motif not being
            at the very end of the sequence. Thus, the 'FN' motifs should not
            be used as an 'arbitrary truth' for the benchmarking purposes.
            <br>
            In one of the sequences BAliBASE annotated the motif as a 
            3-residue segment ("DEL"), but we suppose it was not done on
            purpose.
            </p>
          </div>
        </div>
        <br>
        <strong>Alignment 1</strong>
        Motif score = 0.37
        <br>
        <div class="row">
          <div class="col-md-6" style="overflow:auto;">
            <img src="{{ url_for('static',
            filename="alignments/balibase_issues/pics/kmad_clustal_refine80.png") }}" >
            <br>
          </div>
          <div class="col-md-6" style="overflow:auto;">
            <img src="{{ url_for('static',
            filename="alignments/balibase_issues/pics/kmad_clustal_refine80_KDEL.png") }}" >
          </div>
        </div>
        <hr>
        <strong>Alignment 2</strong>
        Motif score = 0.88
        <br>
        <div class="row">
          <div class="col-md-6" style="overflow:auto;">
            <img src="{{ url_for('static',
            filename="alignments/balibase_issues/pics/kmad_tcof_refine77.png") }}" >
            <br>
          </div>
          <div class="col-md-6" style="overflow:auto;">
            <img src="{{ url_for('static',
            filename="alignments/balibase_issues/pics/kmad_tcof_refine77_KDEL.png") }}" >
            <br>
          </div>
        </div>
        <hr>
        <strong>Alignment 3</strong>
        Motif score = 0.78
        <br>
        <div style="overflow:auto;">
          <img src="{{ url_for('static',
          filename="alignments/balibase_issues/pics/kmad_tcof_refine79.png") }}" >
          <br>
          <img src="{{ url_for('static',
          filename="alignments/balibase_issues/pics/kmad_tcof_refine79_KDEL.png") }}" >
        </div>
        <br>
      </div>
    </div>
  </div>
</div>
<div class="panel panel-default" style="width=100%;" id="panel0">
  <div class="panel-heading">
    <div class="panel-title">
      <strong>RV913_BOX142</strong>
    </div>
  </div>
  <div class="panel-body panel-body-custom">
    <div class="row">
      <div class="col-xs-12 col-md-6" style="overflow:auto;">
        <p align="justify">
        In this alignment the highlighted motif is a C-mannosylation site
        (ELM ID: MOD_CMANNOS). The regular expression ("W..W") doesn't match 
        the 4th
        and 5th sequences. Moreover, the C-mannosylation site is a motif for
        attachment of a mannosyl residue to a tryptophane (first tryptophane
        in the motif) - and in both these sequences the first tryptophane is
        substituted with another amino acid.
        <br>
        <div class="row">
          <div class="col-xs-12 col-md-12">
            Residues annotated in BAliBASE as true positives and/or false
            negatives:
            <br>
            <img src="{{ url_for('static',
            filename="alignments/balibase_issues/pics/RV913_BOX142_bali.png") }}" >
          </div>
          <div class="col-xs-12 col-md-12">
            Segments matching the motif's pattern:
            <br>
            <br>
            <img src="{{ url_for('static',
            filename="alignments/balibase_issues/pics/RV913_BOX142_single.png") }}" >
          </div>
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
<div class="panel panel-default" style="width=100%;" id="panel0">
  <div class="panel-heading">
    <div class="panel-title">
      <strong>RV913_BOX214</strong>
    </div>
  </div>
  <div class="panel-body panel-body-custom">
    <div class="row">
      <div class="col-xs-12 col-md-6" style="overflow:auto;">
        <p align="justify">
        This alignment is a similar case as the previous one. Here the motif
        is an N-glycosylation site (ELM ID: MOD_N-GLC_2).
        Only the first two sequences match the regular expression, "N[^P]C",
        and in the three other sequences the first residue is substituted,
        which is the glycosylated asparagine - therefore, it obviously can 
        no longer be an N-glycosylation site.
        </p>
        <br>
        <div class="row">
          <div class="col-xs-12 col-md-12">
            Residues annotated in BAliBASE as true positives and/or false
            negatives:
            <br>
            <img src="{{ url_for('static',
            filename="alignments/balibase_issues/pics/RV913_BOX214_bali.png") }}" >
          </div>
        </div>
        <hr>
        <div class="row">
          <div class="col-xs-12 col-md-12">
            Segments matching the motif's pattern:
            <br>
            <img src="{{ url_for('static',
            filename="alignments/balibase_issues/pics/RV913_BOX214_single.png") }}" >
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
<div class="panel panel-default" style="width=100%;" id="panel0">
  <div class="panel-heading">
    <div class="panel-title">
      <strong>RV913_BOX146</strong>
    </div>
  </div>
  <div class="panel-body panel-body-custom">
    <div class="row">
      <div class="col-xs-12 col-md-6" style="overflow:auto;">
        <p align="justify">
        The motif annotated here is a clathrin box (ELM ID:
        LIG_Clathr_ClatBox_1). Its regular expression is "[IVLMF].[IVLMF][DE]"
        The motif pattern matches the fourth sequence
        in a different location than annotated by BAliBASE - much further from 
        the C-terminus. 
        Obviously one would not like to use this outlying motif for the motif 
        score, but it is not fair to give penalty points for a 
        an alignment of a motif that is not there. 
        Although BAliBASE alignment 
        is in this case most probably correct, alignment of this segment should
        in our opinion
        be included only in the general score and not in the motif score.
        <br>
        </p>
        <div class="alert alert-warning" role="alert">
          Scroll to the right see the rest of the motifs.
        </div>
      </div>
      <div class="col-xs-12 col-md-12" style="overflow:auto;">
        Residues annotated in BAliBASE as true positives and/or false
        negatives:
        <br>
        <img src="{{ url_for('static',
        filename="alignments/balibase_issues/pics/RV913_BOX146_bali.png") }}" >
        <hr>
        Segments matching the motif's pattern:
        <br>
        <img src="{{ url_for('static',
        filename="alignments/balibase_issues/pics/RV913_BOX146_single.png") }}" >
      </div>
    </div>
  </div>
</div>
<div class="panel panel-default" style="width=100%;" id="panel0">
  <div class="panel-heading">
    <div class="panel-title">
      <strong>RV913_BOX082</strong>
    </div>
  </div>
  <div class="panel-body panel-body-custom">
    <div class="row">
      <div class="col-xs-12 col-md-6" style="overflow:auto;">
        <p align="justify">
        The motif in this example is an SH2 ligand (ELM ID: LIG_SH2_STAT5,
        regex: "(Y)[VLTFIC].."). The pattern does not match the segment in
        the last sequence.
        <br>
        This example represents another problem with the BAliBASE motif
        benchmark - if there are several instances of a motif in an
        alignment it only scores for one instance. Thus an alignment that would
        get two of the motif columns aligned right but the rightmost column
        wrong, would get a lower score than an alignment where only the last
        column is well aligned.
        <br>
        Residues annotated in BAliBASE as true positives and/or false
        negatives:
        <br>
        </p>
      </div>
      <div class="col-xs-12 col-md-12" style="overflow:auto;">
        <img src="{{ url_for('static',
        filename="alignments/balibase_issues/pics/RV913_BOX082_bali.png") }}" >
        <hr>
        Segments matching the motif's pattern:
        <br>
        <img src="{{ url_for('static',
        filename="alignments/balibase_issues/pics/RV913_BOX082_single.png") }}" >
      </div>
    </div>
  </div>
</div>
<div class="panel panel-default" style="width=100%;" id="panel0">
  <div class="panel-heading">
    <div class="panel-title">
      <strong>RV913_BOX036</strong>
    </div>
  </div>
  <div class="panel-body panel-body-custom">
    <div class="row">
      <div class="col-xs-12 col-md-6" style="overflow:auto;">
        <p align="justify">
        Here the motif is a tyrosine phosphorylation site 
        (ELM ID: MOD_TYR_CSK, regex: "[TAD][EA].Q(Y)[QE].[GQA][PEDLS]").
        The pattern does not match three of the sequences - although it would
        match the third sequence if the unknown residue X was a tyrosine.
        <br>
        Residues annotated in BAliBASE as true positives and/or false
        negatives:
        </p>
        <br>
        <img src="{{ url_for('static',
        filename="alignments/balibase_issues/pics/RV913_BOX036_bali.png") }}" >
        <hr>
        Segments matching the motif's pattern:
        <br>
        <img src="{{ url_for('static',
        filename="alignments/balibase_issues/pics/RV913_BOX036_single.png") }}" >
      </div>
    </div>
  </div>
</div>
<div class="panel panel-default" style="width=100%;" id="panel0">
  <div class="panel-heading">
    <div class="panel-title">
      <strong>RV913_BOX034</strong>
    </div>
  </div>
  <div class="panel-body panel-body-custom">
    <div class="row">
      <div class="col-xs-12 col-md-6" style="overflow:auto;">
        <p align="justify">
        The motif in this alignment is a PKA phosphorylation site (MOD_PKA_1).
        In one of the sequences the motif is assigned although it does not
        match the pattern from ELM ("[RK][RK].([ST])[^P]..").
        The modified serine is in this case substituted with a glycine,
        hence our conclusion that it is a true negative and not a false
        negative as BAliBASE says.
        <br>
        Residues annotated in BAliBASE as true positives and/or false
        negatives:
        </p>
        <br>
        <img src="{{ url_for('static',
        filename="alignments/balibase_issues/pics/RV913_BOX034_bali.png") }}" >
        <hr>
        Segments matching the motif's pattern:
        <br>
        <img src="{{ url_for('static',
        filename="alignments/balibase_issues/pics/RV913_BOX034_single.png") }}" >
      </div>
    </div>
  </div>
</div>
<div class="panel panel-default" style="width=100%;" id="panel0">
  <div class="panel-heading">
    <div class="panel-title">
      <strong>RV913_BOX012</strong>
    </div>
  </div>
  <div class="panel-body panel-body-custom">
    <div class="row">
      <div class="col-xs-12 col-md-6" style="overflow:auto;">
        <p align="justify">
        Here only two of the sequence match the SID motif's 
        (ELM ID: LIG_Sin3_2) pattern
        ("[FHYM].A[AV].[VAC]L[MV].[MI]").
        The sequence miss only one residue to match the motif pattern, so of
        course it is possible that overruling the pattern by BAliBASE is a
        wise move. 
        Nevertheless, cases like this are one of the reasons of the noise in the 
        motif scores.
        <br>
        Residues annotated in BAliBASE as true positives and/or false
        negatives:
        </p>
        <br>
        <img src="{{ url_for('static',
        filename="alignments/balibase_issues/pics/RV913_BOX012_bali.png") }}" >
        <hr>
        Segments matching the motif's pattern:
        <br>
        <img src="{{ url_for('static',
        filename="alignments/balibase_issues/pics/RV913_BOX012_single.png") }}" >
      </div>
    </div>
  </div>
</div>
<div class="panel panel-default" style="width=100%;" id="panel0">
  <div class="panel-heading">
    <div class="panel-title">
      <strong>RV913_BOX132</strong>
    </div>
  </div>
  <div class="panel-body panel-body-custom">
    <div class="row">
      <div class="col-xs-12 col-md-6" style="overflow:auto;">
        <p align="justify">
        The motif in this alignment is a PIKK phosphorylation site (ELM ID:
        MOD_PIKK_1), with regular expression "...([ST])Q..".
        The second sequence is does not match the pattern because the
        phosphorylated serine is replaced with an asparagine.
        In this sequence the motif could have 'moved' a couple of residues
        to the left, where the motif pattern is found. It might be another 
        case of residues that are not part of a motif contributing to 
        the motif score, which adds to the noise in the benchmarking results.
        <br>
        Residues annotated in BAliBASE as true positives and/or false
        negatives:
        </p>
        <br>
        <img src="{{ url_for('static',
        filename="alignments/balibase_issues/pics/RV913_BOX132_bali.png") }}" >
        <hr>
        Segments matching the motif's pattern:
        <br>
        <img src="{{ url_for('static',
        filename="alignments/balibase_issues/pics/RV913_BOX132_single.png") }}" >
      </div>
    </div>
  </div>
</div>
<div class="panel panel-default" style="width=100%;" id="panel0">
  <div class="panel-heading">
    <div class="panel-title">
      <strong>RV913_BOX290</strong>
    </div>
  </div>
  <div class="panel-body panel-body-custom">
    <div class="row">
      <div class="col-xs-12 col-md-6" style="overflow:auto;">
        <p align="justify">
        The motif in this alignment is a PTS1 motif (ELM ID: TRG_PTS1). 
        In the third sequence it
        does not match the ELM pattern
        ("(.[SAPTC][KRH][LMFI]$)|([KRH][SAPTC][NTS][LMFI]$)").
        <br>
        Residues annotated in BAliBASE as true positives and/or false
        negatives:
        </p>
        <br>
        <img src="{{ url_for('static',
        filename="alignments/balibase_issues/pics/RV913_BOX290_bali.png") }}" >
        <hr>
        Segments matching the motif's pattern:
        <br>
        <img src="{{ url_for('static',
        filename="alignments/balibase_issues/pics/RV913_BOX290_single.png") }}" >
      </div>
    </div>
  </div>
</div>
</div>
{% endblock %}
{% block js %}
<script>
</script>
{% endblock %}
