{% extends "base.html" %}
{% block additional_imports %}
    <link rel="stylesheet" href="{{ url_for('static', filename='css/methods.css') }}">
{% endblock %}
{% block content %}
<div class="panel-group" id="panel-group0">
  <div class="panel panel-default" style="width=100%;" id="panel0">
    <div class="panel-heading">
      <h4 class="panel-title">
        <span class="panel_title_text"><strong>Disorder prediction</strong></span>
       </h4>
    </div>
    <div class="panel-body">
      <span class="panel_body_text">
        <p>  
        We first try to find your sequence in the
        <strong>D<sup>2</sup>P<sup>2</sup></strong> 
      database of precomputed disorder predictions. 
      If the sequence is not present there, the 
      disorder prediction is performed with 4 different predictors:
      <strong>DISOPRED</strong>, <strong>SPINE-D</strong>, <strong>PSIPRED</strong>, and
      <strong>PreDisorder</strong>.
      </p>
      </br>
      <p>
      <strong>D<sup>2</sup>P<sup>2</sup></strong> <br>
      Database of Disordered Protein Predictions
      <br>
      <br>
       Oates,M.E., Romero,P., Ishida,T., Ghalwash,M., Mizianty,M.J., Xue,B.,
       Dosztányi,S., Uversky,V.N., Obradovic,Z., Kurgan,L., Dunker,A.K.,
       Gough,J. (2013). "D2P2: Database of Disordered Protein Predictions."
       <em>Nucleic Acids Research</em>
       <strong>41</strong>
       (D1):D508-D516
       <br>
       <br>
       <strong>DISOPRED</strong>
       <br>
       <br>
       Ward JJ, Sodhi JS, McGuffin LJ, Buxton BF and Jones DT (2004)
       <br>
       Prediction and functional analysis of native disorder in proteins from
       the three kingdoms of life
       <br>
       <i>Journal of Molecular Biology</i>
       , <b>337</b>:635-645. 
       <br>
       <br>
       <strong>SPINE-D</strong>
       <br>
       <br>
        Tuo Zhang, Eshel Faraggi, Bin Xue, A. Keith Dunker, Vladimir N.
        Uversky, Yaoqi Zhou. SPINE-D: Accurate prediction of short and long
        disordered regions by a single neural-network based method.
        <i> Journal of Biomolecular Structure and Dynamics.</i>
        2012,
        <b>29</b>
        (4):799-813.
        <br>
        <br>
        <strong>PSIPRED</strong>
        <br>
        <br>
        Jones, D.T. (1999) Protein secondary structure prediction based on
        position-specific scoring matrices. <i>Journal of Molecular Biology</i>
        <b>292</b>:195-202.
        <br>
        <br>
        <strong>PreDisorder</strong>
        <br>
        <br>
        X. Deng, J. Eickholt, and J. Cheng. PreDisorder: Ab Initio
        Sequence-Based Prediction of Protein Disordered Regions. <i>BMC
          Bioinformatics</i>, <b>10</b>:436, 2009.
        <br>
        J. Hecker, J. Yang, and J. Cheng. Protein Disorder Prediction at
        Multiple Levels of Sensitivity and Specificity. <i>BMC Genomics.</i> <b>9</b>(S1):S9, 2008.
        <br>
        J. Cheng, M. Sweredoski, and P. Baldi. Accurate Prediction of Protein
        Disordered Regions by Mining Protein Structure Data, <i>Data Mining and
        Knowledge Discovery</i>, vol. 11, no. 3, pp. 213-222, 2005.
      </p>
      </span>
    </div>
  </div>
  <div class="panel panel-default" style="width=100%;" id="panel1">
    <div class="panel-heading">
      <h4 class="panel-title">
        <span class="kman-k">K</span><span class="kman-m">M</span><span
            class="kman-a">A</span><span class="kman-n">N</span> 
            <br>
            <span
              class="kman-k">K</span><span class="panel_title_text">nowledge
              Based </span><span
              class="kman-m">M</span><span class="panel_title_text">ultiple
              Sequence </span><span
              class="kman-a">A</span><span class="panel_title_text">lignment
              for </span><span
              class="kman-n">N</span><span class="panel_title_text">atively
              Unfolded Proteins </span>
      </h4>
    </div>
    <div class="panel-body">
      <span class="panel_body_text">
        <p>
        The tool incorporates information from databases and predictions about
        functional sites to perform a Multiple Sequence Alignment (MSA).
        The information includes phosphorylations predicted by <strong>NetPhos</strong>, annotated
        posttranslational modifications (PTMs) from <strong>Uniprot</strong>, short linear
        motifs (SLiMs) from <strong>ELM</strong>, and functional domains
        predicted by <strong>Pfam</strong>.
        </p>
        <br>
        <p>
        <strong>ELM</strong>
        <br>
        <br>
        Dinkel H., Van Roey K., Michael S., Davey N.E., Weatheritt R.J., Born D., Speck T., Krüger D.,
        Grebnev G., Kubań M., Strumiłło M., Uyar B., Budd A., Altenberg B., Seiler M., Chemes L.B.,
        Glavina J., Sánchez I.E., Diella F., Gibson T.J.
        The eukaryotic linear motif resource ELM: 10 years and counting
        <i>Nucleic Acids Research</i> (2013) Database issue 
        <b>42</b>:D259-266
        <br>
        <br>
        <strong>NetPhos</strong>
        <br>
        <br>
        Blom, N., Gammeltoft, S., and Brunak, S.
        Sequence- and structure-based prediction of eukaryotic protein
        phosphorylation sites.
        <i>Journal of Molecular Biology</i>: <b>294</b>(5): 1351-1362, 1999. 
        <br>
        <br>
        <strong>Pfam</strong>
        <br>
        <br>
        R.D. Finn, A. Bateman, J. Clements, P. Coggill, R.Y. Eberhardt, S.R.
        Eddy, A. Heger, K. Hetherington, L. Holm, J. Mistry, E.L.L. Sonnhammer,
        J. Tate, M. Punta. The Pfam protein families database <i>Nucleic Acids Research</i> (2014)  Database Issue
        <b>42</b>:D222-D230 
        <br>
        <br>
        <strong>Uniprot</strong>
        <br>
        <br>
        The UniProt Consortium.
        Activities at the Universal Protein Resource (UniProt)
        <i>Nucleic Acids Res.</i> <b>42</b>: D191-D198 (2014).
        <br>
        <br>
        </p>
      </span>
    </div>
  </div>
</div>
{% endblock %}
{% block js %}
{% endblock %}
